pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -179,8 +176,40 @@ func (pbs *pullthroughBlobStore) copyContent(store BlobGetterService, ctx contex
 	return desc, nil
 }
 
+// storeLocalInBackground spawns a separate thread to copy the remote blob from the remote registry to the
+// local blob store.
+// The function assumes that localBlobStore is thread-safe.
+func storeLocalInBackground(ctx context.Context, repo *repository, localBlobStore distribution.BlobStore, dgst digest.Digest) {
+	// leave only the essential entries in the context
+	newCtx := context.WithValue(context.Background(), "instance.id", ctx.Value("instance.id"))
```

Comment:
> docker/distribution/context.Background already provides "instance.id".

My intention was to make it possible to associate this job with the original request, but a better way to do that would probably be based on `http.request.id` instead, which is already logged by the logger.

I'll ditch the `instance.id` and `instance.jobid` and let the context generate new `instance.id` instead.

> I would like to preserve "trace.id", although we don't use distribution/context.WithTrace yet.

We don't use it yet so I don't see a point in preserving it. Once we start to use it, this function will set the `trace.*` attributes on its own anyway.